.github/workflows/check.yml

```diff
@@ -7,6 +7,26 @@ env:
   TESTS_TIMEOUT: 10 # in minutes
 
 jobs:
+  pre-commit:
+    name: pre-commit sanity checks
+    runs-on: ubuntu-22.04
+
+    steps:
+    - name: Install dependencies
+      run: |
+        sudo apt-get update --quiet
+        sudo apt-get upgrade --quiet -y
+        sudo apt-get install --quiet -y --no-install-recommends python3-pip git
+
+    - name: Check out xdg-desktop-portal
+      uses: actions/checkout@v3
+
+    - name: Install pre-commit
+      run: pip install pre-commit
+
+    - name: Run pre-commit checks
+      run: pre-commit run --all
+
```

Comment:
Using https://pre-commit.ci/ would be a lot simpler, and it runs a lot faster.

Reply:
Thanks, that sounds like a plan but ... how do we actually use that? I'm staring at that website but I can't find any guidance on how to enable this.

Reply:
You have to click on the "Sign in with GitHub" button. I guess it needs to be done by someone with permissions to enable it for this repository.